cartography/intel/gcp/compute.py

```diff
@@ -0,0 +1,118 @@
+# Google Compute Engine API-centric functions
+# https://cloud.google.com/compute/docs/concepts
+from googleapiclient.discovery import HttpError
+import json
+import logging
+
+from cartography.util import run_cleanup_job
+from cartography.intel.gcp.crm import get_gcp_projects
+
+logger = logging.getLogger(__name__)
+
+
+def get_zones_in_project(project_id, compute, max_results=None):
+    """
+    Return the zones where the Compute Engine API is enabled for the given project_id.
+    If the API is not enabled, return None.
+    :param project_id: The project id
+    :param compute: The compute resource object created by googleapiclient.discovery.build()
+    :param max_results: Optional cap on number of results returned by this function. Default = None, which means no cap.
+    :return: List of a project's zone objects if Compute API is turned on, else None.
+    """
+    try:
+        req = compute.zones().list(project=project_id, maxResults=max_results)
+        res = req.execute()
+        return res['items']
+    except HttpError as e:
+        reason = json.loads(e.content)['error']['errors'][0]['reason']
+        if reason == 'accessNotConfigured':
+            logger.debug(
+                (
+                    "Google Compute Engine API access is not configured for project %s. "
+                    "Full details: %s"
+                ),
+                project_id,
+                e
+            )
+            return None
```

Comment:
How does the user know the user doesn't have API access? should we bubble this up to the CLI vs returning None?

Reply:
This `logger.debug` call and the Google SDK's built-in telemetry logs a debug message to the CLI.  Google's message looks like this:
`WARNING:googleapiclient.http:Encountered 403 Forbidden with reason "accessNotConfigured"
INFO:googleapiclient.discovery:URL being requested: GET https://www.googleapis.com/compute/v1/projects/<project_name>/zones?alt=json`